src/main/java/ru/yandex/practicum/filmorate/dao/film/FilmDaoImpl.java

```diff
@@ -196,4 +343,32 @@ private Genre getGenreById(Long id) {
         return jdbcTemplate.queryForObject("select * from genres where id = ?", (rs, rowNum) ->
                 new Genre(rs.getLong("id"), rs.getString("name")), id);
     }
+
+    @Override
+    public List<Film> getRecommendations(Long id) {
+        String sqlGetFilmsIdByUserId = "select film_id from films_users where user_id=?";
+        List<Long> idFilmsOfOurUser = jdbcTemplate.query(sqlGetFilmsIdByUserId, (rs, rowNum) -> rs.getLong("film_id"), id);
+        if (idFilmsOfOurUser.isEmpty()) {
+            log.info("Не найдено фильмов для пользователя с идентификатором {}", id);
+            return Collections.emptyList();
+        }
+        String sqlAllUsersIdWhoLikedSameFilms = "select distinct user_id from films_users where film_id in " +
+                "(" + sqlGetFilmsIdByUserId + ") and user_id != ?";
+        log.info("Поиск пользователей, которым понравились те же фильмы, что и пользователю с идентификатором {}", id);
+        List<Long> usersId = jdbcTemplate.query(sqlAllUsersIdWhoLikedSameFilms, (rs, rowNum) -> rs.getLong("user_id"), id, id);
+        if (usersId.isEmpty()) {
+            return Collections.emptyList();
+        }
+        Set<Film> data = new HashSet<>();
+        for (Long userId : usersId) {
+            List<Long> userFilmsId = jdbcTemplate.query(sqlGetFilmsIdByUserId, (rs, rowNum) -> rs.getLong("film_id"), userId);
+            for (Long filmId : userFilmsId) {
+                if (!idFilmsOfOurUser.contains(filmId)) {
+                    data.add(getFilmById(filmId));
+                }
+            }
+        }
+        log.info("Рекомендации, сгенерированные для пользователя с идентификатором {}", id);
+        return data.stream().collect(Collectors.toList());
```

Comment:
Лучше использовать конструктор
```suggestion
        return new ArrayList<>(data);
```